src/Microsoft.Extensions.Configuration/ConfigurationProvider.cs

```diff
@@ -53,10 +58,40 @@ public virtual void Set(string key, string value)
         /// <summary>
         /// Loads (or reloads) the data for this provider.
         /// </summary>
-        public virtual void Load()
+        public void Load()
         {
+            try
+            {
+                DoLoad();
+            }
+            catch (Exception e)
+            {
+                bool ignoreException = false;
+                if (OnLoadException != null)
+                {
+                    var exceptionContext = new LoadExceptionContext
+                    {
+                        Provider = this,
+                        Exception = e
+                    };
+                    OnLoadException.Invoke(exceptionContext);
+                    ignoreException = exceptionContext.Ignore;
+                }
+                if (!ignoreException)
+                {
+                    throw e;
+                }
+            }
         }
-       
+
+        /// <summary>
+        /// Loads (or reloads) the data for this provider.
+        /// </summary>
+        public virtual void DoLoad()
```

Comment:
Today Load isn't abstract since most of our implementations actually do nothing.  So a no-op DoLoad was parity, my first feeling was that it should be abstract too :)